server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -478,66 +513,144 @@ private File compileMain(File maincpp, Map<String, Object> manifestContext) thro
+        File[] srcDirs = {srcDir};
+        return listFiles(srcDirs, regEx);
+    }
 
-        File[] srcFiles = null;
-        if (srcDir.isDirectory()) {
-            Collection<File> _srcFiles = new ArrayList<>();
-            _srcFiles = FileUtils.listFiles(srcDir, null, true);
-            _srcFiles = Extender.filterFiles(_srcFiles, platformConfig.sourceRe);
+    private List<File> generateProtoCxxForEngine(File extDir, Map<String, Object> manifestContext, List<File> protoFiles) throws IOException, InterruptedException, ExtenderException {
+        List<File> generated = new ArrayList<>();
 
-            // sorting makes it easier to diff different builds
-            srcFiles = _srcFiles.toArray(new File[_srcFiles.size()]);
+        Map<String, Object> tmpcontext = context(manifestContext);
 
+        if (commonPlatformConfig == null || commonPlatformConfig.protoEngineCxxCmd == null) {
+            LOGGER.info("Not supporting .proto files");
+            return generated;
         }
 
-        if (srcFiles == null) {
+        LOGGER.info("Converting .proto to engine .cpp files for extension {}", extDir.getName());
+
+        File extBuildDir = createDir(buildDirectory, extDir.getName());
+
+        List<String> includes = getIncludeDirs(extDir);
+
+        for (File protoFile : protoFiles) {
+
+            String name = ExtenderUtil.switchExtension(protoFile.getName(), ".cpp");
+            File tgtCpp = new File(extBuildDir, name);
+
+            Map<String, Object> context = context(manifestContext);
+
+            context.put("src", ExtenderUtil.getRelativePath(jobDirectory, protoFile));
+            context.put("ext", ImmutableMap.of("includes", includes));
+            context.put("out_dir", extBuildDir);
+
+            String command = templateExecutor.execute(commonPlatformConfig.protoEngineCxxCmd, context);
+            processExecutor.execute(command);
+
+            LOGGER.info("Generated {}", tgtCpp);
+
+            generated.add(tgtCpp);
+        }
+        return generated;
+    }
+
+    private List<File> generateProtoSrcForPlugin(File extDir, Map<String, Object> manifestContext, List<File> protoFiles, String language) throws IOException, InterruptedException, ExtenderException {
+        List<File> generated = new ArrayList<>();
+
+        Map<String, Object> tmpcontext = context(manifestContext);
+
+        File extBuildDir = createDir(buildDirectory, extDir.getName());
+        List<String> includes = getIncludeDirs(extDir);
+
+        LOGGER.info("Converting .proto to plugin {} files for extension {}", language, extDir.getName());
+
+        String suffix = "";
+        if (language.equals("cpp"))
+            suffix = ".pb.cc";
+        else if (language.equals("java"))
+            suffix = ".java";
+        else if (language.equals("python"))
+            suffix = "_pb2.py";
+        else
+            throw new ExtenderException("Proto output doesn't support language: " + (language!=null?language:"null"));
```

Comment:
These files are for manipulating source/output content.
We can convert .proto into files for cpp, java and python.